src/states/joining_peer.rs

```diff
@@ -254,51 +463,38 @@ impl Base for JoiningPeer {
         msg: Message,
         _outbox: &mut dyn EventBox,
     ) -> Result<Transition, RoutingError> {
+        // Ignore messages from peers we didn't send `BootstrapRequest` to.
+        if let Stage::Bootstrapping(stage) = &self.stage {
+            let sender = msg.src.to_sender_node(sender)?;
+
+            if !stage.pending_requests.contains(sender.peer_addr()) {
+                debug!(
+                    "{} - Ignoring message from unexpected peer: {}: {:?}",
+                    self, sender, msg,
+                );
+                self.network_service.disconnect(*sender.peer_addr());
```

Comment:
it's not only `ignore messages` but also disconnect from it?

Reply:
Yes, this is because of the implicit connection that quic-p2p does, to avoid keeping unneeded connections.

Reply:
Then, I mean shall the log mention the disconnection as well?